lib/rules/prefer-const.js

```diff
@@ -240,11 +259,32 @@ module.exports = {
          * @returns {void}
          */
         function report(node) {
-            context.report({
-                node: node,
-                message: "'{{name}}' is never reassigned, use 'const' instead.",
-                data: node
-            });
+            var reportArgs = {
+                    node: node,
+                    message: "'{{name}}' is never reassigned, use 'const' instead.",
+                    data: node
+                },
+                isNormalVariableDeclaration = node.parent.parent.type === "VariableDeclaration" &&
+                    (node.parent.parent.parent.type === "ForInStatement" ||
+                        node.parent.parent.parent.type === "ForOfStatement" ||
+                        node.parent.parent.type === "ForStatement" ||
+                        node.parent.init),
+                isDestructuring = node.parent.parent.type === "ObjectPattern" &&
+                    node.parent.parent.parent.parent.type === "VariableDeclaration",
```

Comment:
Thank you, but I'm still not sure.
Why do you fix `let {a, b, c}`, but not `let [a, b, c]` or `let {a: {b, c}}` ?
Also, the variable name `isDestructuring` is not expressing this condition.